modules/300-prometheus/hooks/prometheus_disk.go

```diff
@@ -0,0 +1,485 @@
+		if status.Name == "prometheus" {
+			containerReady = status.Ready
+			break
+		}
+	}
+
+	return PodFilter{
+		Name:           pod.Name,
+		Namespace:      pod.Namespace,
+		PodScheduled:   podScheduled,
+		ContainerReady: containerReady,
+		PromName:       pod.Labels["prometheus"],
+	}, nil
+}
+
+func prometheusDisk(input *go_hook.HookInput, dc dependency.Container) error {
+	// checking pvc status for FileSystemResizePending or Resizing and restarting the pod if needed
+	podDeletionFlag := false
+	for _, obj := range input.Snapshots["pvcs"] {
+		pvc := obj.(PersistentVolumeClaimFilter)
+		if pvc.ResizePending {
+			podName := fmt.Sprintf("prometheus-%s-%s", pvc.PromName, pvc.Name[len(pvc.Name)-1:])
+			for _, pObj := range input.Snapshots["pods"] {
+				pod := pObj.(PodFilter)
+				if podName == pod.Name && pod.PodScheduled {
+					input.LogEntry.Infof("PersistentVolumeClaim %s in FileSystemResizePending state, deletion pod %s to complete resize", pvc.Name, podName)
+					// "300-prometheus/hooks/prometheus_disk.go Module hook failed, requeue task to retry after delay. Failed count is 1. Error: 1 error occurred:\n\t* Delete object v1/Pod/d8-monitoring/prometheus-main-0: timed out waiting for the condition\n\n"
+					input.PatchCollector.Delete("v1", "Pod", "d8-monitoring", podName)
+					podDeletionFlag = true
+				}
+			}
+		}
+	}
+
+	if podDeletionFlag {
+		return nil
+	}
+
+	proms := []string{"main", "longterm"}
+	for _, promName := range proms {
+
+		promNameForPath := strings.ToUpper(promName[0:1]) + promName[1:]
+
+		var diskSize int64  // GiB
+		var retention int64 // GiB
+
+		pvcExists := false
+		for _, obj := range input.Snapshots["pvcs"] {
+			if obj.(PersistentVolumeClaimFilter).PromName == promName {
+				pvcExists = true
+				break
+			}
+		}
+
+		// if there is no PVC, set the default values for diskSize and retention
+		if !pvcExists {
+			effectiveStorageClass := input.Values.Get(fmt.Sprintf("prometheus.internal.prometheus%s.effectiveStorageClass", promNameForPath)).String()
+			input.LogEntry.Infof("prometheus.internal.prometheus%s.effectiveStorageClass: %s", promNameForPath, effectiveStorageClass)
+			if effectiveStorageClass != "false" && isVolumeExpansionAllowed(input, effectiveStorageClass) {
+				diskSize = 25
+				retention = 22
+			} else {
+				diskSize = 30
+				retention = 27
+			}
+			//	otherwise, we calculate diskSize and retention
+		} else {
+			diskResizeLimit := int64(300)
+			maxDiskSizeConfigPath := fmt.Sprintf("prometheus.%sMaxDiskSizeGigabytes", promName)
+			if input.ConfigValues.Exists(maxDiskSizeConfigPath) {
+				diskResizeLimit = input.ConfigValues.Get(maxDiskSizeConfigPath).Int()
+			}
+
+			desiredSize := calcDesiredSize(input, dc, promName)
+
+			if desiredSize <= diskResizeLimit {
+				diskSize = desiredSize
+			}
+
+			for _, obj := range input.Snapshots["pvcs"] {
+				pvc := obj.(PersistentVolumeClaimFilter)
+				if pvc.PromName == promName {
```

Comment:
It would help if you tried to avoid nesting clauses and increasing indentation.

```go
if pvc.PromName != promName {
    continue
}

if diskSize == 0 && pvc.RequestsStorage >= diskSize {
    diskSize = pvc.RequestsStorage
    continue
}

patch := composeDiskSizePatch(diskSize)

input.LogEntry.Infof("PersistentVolumeClaim %s size will be changed from %dGB to %dGB", pvc.Name, pvc.RequestsStorage, diskSize)
input.PatchCollector.MergePatch(patch, "v1", "PersistentVolumeClaim", "d8-monitoring", pvc.Name)
```

Reply:
It is easier to read code if most lines start on the same level.

You can search a little more about indentation complexity on the internet
https://holgerfrohloff.de/indentation-as-a-complexity-metric

Reply:
Fixed.